realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java

```diff
@@ -443,18 +433,20 @@ protected final void copy(ColumnInfo rawSrc, ColumnInfo rawDst) {
 
         proxyState.getRealm$realm().checkIfValid();
         OsList osList = proxyState.getRow$realm().getModelList(columnInfo.columnRealmListIndex);
-        osList.removeAll();
-        if (value == null) {
-            return;
-        }
-        for (RealmModel linkedObject : value) {
-            if (!(RealmObject.isManaged(linkedObject) && RealmObject.isValid(linkedObject))) {
-                throw new IllegalArgumentException("Each element of 'value' must be a valid managed object.");
+        if (value != null && value.size() == osList.size()) {
+            for (int i = 0; i < value.size(); i++) {
+                some.test.AllTypes linkedObject = value.get(i);
+                osList.setRow(i, ((RealmObjectProxy) linkedObject).realmGet$proxyState().getRow$realm().getIndex());
```

Comment:
As said before, shouldn't we check the `proxyState.checkValidObject(linkedObject);` the previous (deleted code did this check as you can see)

Reply:
Yes, good catch 👍